app/controllers/search_controller.rb

```diff
@@ -57,10 +57,16 @@ def tags
 
   def chat_channels
     ccm_docs = Search::ChatChannelMembership.search_documents(
-      params: chat_channel_params.merge(user_id: current_user.id).to_h,
+      params: chat_channel_params.merge(user_id: chat_channel_params[:user_id] || current_user.id).to_h,
```

Comment:
If we are going to allow this as a param we should validate that it is either `all` or the current_user ID otherwise someone could theoretically fetch another user's chat channels